src/replays_utils.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const { DemoParser } = require('sdfz-demo-parser');
+
+// Parse a replay and create the needed info structure. If there is a cache
+// file for this replay, use it.
+async function parseReplay(springPath, replayPath, exitIfCacheExists=true) {
+
+	const fullReplayPath = path.join(springPath, replayPath);
+	const demoCachePath = `${fullReplayPath}.cache`;
+
+	// If there is a cache file, return its content (except if user
+	// explicitly asked to bail out in that case)
+	if (fs.existsSync(demoCachePath)) {
+		if (exitIfCacheExists) return null;
+		return JSON.parse(fs.readFileSync(demoCachePath));
+	}
+
+	// Else, go parse the original replay, create the cache file, and
+	// return the info.
+	const parser = new DemoParser();
+	const demo = await parser.parseDemo(fullReplayPath);
+	const info = {
+		relativePath : replayPath,
+		engine: demo.header.versionString,
+		game: demo.info.hostSettings.gametype,
+		map: demo.info.hostSettings.mapname,
+		players: demo.info.players.concat(demo.info.ais),
+		gameTime: demo.header.gameTime
+	};
+
+	// Since there is no cache file yet, create it
+	fs.writeFileSync(demoCachePath, JSON.stringify(info));
+
+	return info;
+}
```

Comment:
can you do all fs. calls here as async? perhaps best to use `require('fs').promises` instead of the plain `require('fs')` 
then it would truly be able to take advantage of async IO

with that I don't think you need a worker thread.

something like this is fine imo:
https://github.com/gajop/spring-launcher/blob/master/src/launcher_downloader.js#L75-L90
(alternatively a global async() function)